UnifiedApi/Worker/Grid/TaskSubmitterWorkerService.cs

```diff
@@ -0,0 +1,387 @@
+  /// </summary>
+  /// <param name="sessionContext">
+  ///   Holds all information on the state of the session at the start of the execution such as session ID.
+  /// </param>
+  /// <param name="taskContext">
+  ///   Holds all information on the state of the task such as the task ID and the payload.
+  /// </param>
+  public byte[] OnInvoke(SessionContext sessionContext,
+                         TaskContext    taskContext)
+    => throw new NotImplementedException("BaseService call OnInvoke won't be implemented");
+
+
+  /// <summary>
+  ///   The middleware triggers the invocation of this handler to unbind the Service Instance from its owning Session.
+  ///   This handler should do any cleanup for any resources that were used in the onSessionEnter() method.
+  /// </summary>
+  /// <param name="sessionContext">
+  ///   Holds all information on the state of the session at the start of the execution such as session ID.
+  /// </param>
+  public void OnSessionLeave(SessionContext sessionContext)
+  {
+  }
+
+
+  /// <summary>
+  ///   The middleware triggers the invocation of this handler just before a Service Instance is destroyed.
+  ///   This handler should do any cleanup for any resources that were used in the onCreateService() method.
+  /// </summary>
+  /// <param name="serviceContext">
+  ///   Holds all information on the state of the service at the start of the execution.
+  /// </param>
+  public void OnDestroyService(ServiceContext serviceContext)
+  {
+  }
+
+  /// <summary>
+  ///   User method to submit task from the service
+  /// </summary>
+  /// <param name="payloads">
+  ///   The user payload list to execute. Generally used for subTasking.
+  /// </param>
+  public IEnumerable<string> SubmitTasks(IEnumerable<byte[]> payloads)
+    => SessionService.SubmitTasks(payloads);
+
+
+  /// <summary>
+  ///   The method to submit several tasks with dependencies tasks. This task will wait for
+  ///   to start until all dependencies are completed successfully
+  /// </summary>
+  /// <param name="payloadWithDependencies">A list of Tuple(taskId, Payload) in dependence of those created tasks</param>
+  /// <param name="resultForParent">Up result to parent task</param>
+  /// <returns>return a list of taskIds of the created tasks </returns>
+  public IEnumerable<string> SubmitTasksWithDependencies(IEnumerable<Tuple<byte[], IList<string>>> payloadWithDependencies,
+                                                         bool                                      resultForParent = false)
+    => SessionService.SubmitTasksWithDependencies(payloadWithDependencies,
+                                                  resultForParent);
+
+
+  /// <summary>
+  ///   User method to submit task from the service
+  /// </summary>
+  /// <param name="serviceContainerBase"></param>
+  /// <param name="payload">
+  ///   The user payload to execute. Generally used for subtasking.
+  /// </param>
+  public string SubmitTask(byte[] payload)
```

Comment:
Is was indeed in base service as extension
ex: public static string SubmitTask<T>(this BaseService<T> serviceContainerBase,
as discussed it can maybe added later on an interface instead of templated class